ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DataColumnSidecarRecoveringCustodyImpl.java

```diff
@@ -235,59 +242,53 @@ private boolean isActiveSuperNode(final UInt64 slot) {
-            .filter(Optional::isPresent)
-            .map(Optional::get)
-            .toList();
-    initiateRecovery(task.block().get(), list);
+    final MetricsHistogram.Timer timer = dataAvailabilityReconstructionTimeSeconds.startTimer();
+    try {
+      initiateRecovery(task, task.existingSidecars.values(), timer);
+    } catch (final Exception e) {
+      task.recoveryStarted.compareAndSet(true, false);
+      LOG.debug(
+          "Error during recovery of {} task with {} sidecars",
+          task.block.get() == null ? "<unknown>" : task.block.get().getSlotAndBlockRoot(),
+          task.existingSidecars.size(),
+          e);
+    }
+    timer.closeUnchecked().run();
   }
 
   private void initiateRecovery(
-      final BeaconBlock block, final SafeFuture<List<DataColumnSidecar>> list) {
-    LOG.debug("Starting data columns sidecars recovery for block: {}", block.getSlotAndBlockRoot());
-
-    final MetricsHistogram.Timer timer = dataAvailabilityReconstructionTimeSeconds.startTimer();
-
-    list.thenAccept(
-            sidecars -> {
-              LOG.debug(
-                  "Recovery for block: {}. DataColumnSidecars found: {}",
-                  block.getSlotAndBlockRoot(),
-                  sidecars.size());
-              final List<DataColumnSidecar> recoveredSidecars =
-                  miscHelpers.reconstructAllDataColumnSidecars(sidecars, kzg);
-              timer.closeUnchecked();
-
-              final Set<UInt64> existingSidecarsIndices =
-                  sidecars.stream()
-                      .map(DataColumnSidecar::getIndex)
-                      .collect(Collectors.toUnmodifiableSet());
-              totalDataAvailabilityReconstructedColumns.inc(
-                  recoveredSidecars.size() - sidecars.size());
-              recoveredSidecars.stream()
-                  .filter(sidecar -> !existingSidecarsIndices.contains(sidecar.getIndex()))
-                  .forEach(
-                      dataColumnSidecar -> {
-                        validDataColumnSidecarsSubscribers.forEach(
-                            l -> l.onNewValidSidecar(dataColumnSidecar, RemoteOrigin.RECOVERED));
-                        delegate
-                            .onNewValidatedDataColumnSidecar(dataColumnSidecar)
-                            .finishError(LOG);
-                        dataColumnSidecarPublisher.accept(dataColumnSidecar);
-                      });
-              LOG.debug(
-                  "Data column sidecars recovery finished for block: {}",
-                  block.getSlotAndBlockRoot());
-            })
-        .alwaysRun(timer.closeUnchecked())
-        .finishError(LOG);
+      final RecoveryTask recoveryTask,
+      final Collection<DataColumnSidecar> sidecars,
+      final MetricsHistogram.Timer timer) {
+    final BeaconBlock block = recoveryTask.block.get();
+
+    LOG.debug(
+        "Recovery for block: {}. DataColumnSidecars found: {}",
+        block.getSlotAndBlockRoot(),
```

Comment:
so theres no guarantee block will be set... i think we should change the `RecoveryTask` object, but i'll expand on that closer to the object.